src/main/java/com/aliyun/openservices/log/Client.java

```diff
@@ -1068,10 +1068,7 @@ public GetCursorResponse GetCursor(GetCursorRequest request)
 		String shardId = String.valueOf(request.GetShardId());
 
 		Map<String, String> headParameter = GetCommonHeadPara(project);
-		StringBuilder resourceUriBuilder = new StringBuilder();
-		resourceUriBuilder.append("/logstores/").append(logStore)
-				.append("/shards/").append(shardId);
-		String resourceUri = resourceUriBuilder.toString();
+        String resourceUri = "/logstores/" + logStore + "/shards/" + shardId;
 

```

Comment:
Which JVM version done this for us?

Reply:
Since 1.5 possibly , see https://stackoverflow.com/questions/1532461/stringbuilder-vs-string-concatenation-in-tostring-in-java